python/ray/autoscaler/updater.py

```diff
@@ -181,10 +181,12 @@ def do_sync(remote_path, local_path, allow_non_existing_paths=False):
 
             with LogTimer(self.log_prefix +
                           "Synced {} to {}".format(local_path, remote_path)):
-                self.cmd_runner.run(
-                    "mkdir -p {}".format(os.path.dirname(remote_path)),
-                    run_env="host")
-                sync_cmd(local_path, remote_path)
+                if not isinstance(self.cmd_runner, DockerCommandRunner):
+                    # The DockerCommandRunner handles this internally
+                    self.cmd_runner.run(
+                        "mkdir -p {}".format(os.path.dirname(remote_path)),
+                        run_env="host")
+                sync_cmd(local_path, remote_path, file_mount=True)
```

Comment:
can we just support like `sync_cmd(local_path, remote_path, options)` or something? having file_mounts be a generic argument but only used for DockerCommandRunner is too leaky for me.

Generally, we should avoid using too many isinstance cases in this part of the code.

Reply:
Totally makes sense!